### ISSUE TITLE
Allow cross origin request for `/nodeinfo/2.0` API

### ISSUE THREAD

**Issue by sxzz** [mastodon/mastodon#23135]
### Steps to reproduce the problem

- Run `fetch('https://mas.to/nodeinfo/2.0')` in your browser console.

```
Access to fetch at 'https://mas.to/nodeinfo/2.0' from origin 'https://elk.zone' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

### Expected behaviour

With CORS header: `access-control-allow-origin: *`

### Actual behaviour

No CORS header.

### Detailed description

There is multiple Mastodon-compatible software like GoToSocial. I want to get the software name of the website in the browser to figure out which software the website used.

### Specifications

Mastodon 4.0.2

**Comment by obale** [comment]
I second that. Additionally, it would be good to set the CORS header for ActivityPub endpoints to allow better interoperability and client to server communication.

Happy to create a new issue with all the necessary endpoints and provide a pull requests. Not familiar with the code base, but what I have seen it should be just a configuration issue.
